shoop/admin/modules/orders/views/shipment.py

```diff
@@ -17,15 +19,68 @@
 from shoop.admin.toolbar import PostActionButton, Toolbar
 from shoop.admin.utils.forms import add_form_errors_as_messages
 from shoop.admin.utils.urls import get_model_url
+from shoop.apps.provides import get_provide_objects
 from shoop.core.excs import NoProductsToShipException
-from shoop.core.models import Order, Product, Supplier
+from shoop.core.models import Order, Product, Shipment, Supplier
+
+FORM_MODIFIER_PROVIDER_KEY = "admin_extend_create_shipment_form"
+
+
+class ShipmentFormModifier(six.with_metaclass(abc.ABCMeta)):
+    def get_extra_fields(self, order):
+        """
+        Extra fields for shipment creation form
+
+        :param order: Order linked to form
+        :type order: shoop.core.models.Order
+        :return: List of extra fields that should be added to form.
+        Tuple should contain field name and Django form field.
+        :rtype: list[tuple(str,django.forms.Field)]
+        """
+        pass
+
+    def clean_hook(self, cleaned_data):
+        """
+        Extra clean for shipment creation form
+
+        :param cleaned_data: form cleaned data
+        :type cleaned_data: dict
+        :return: List of errors that should be handled.
+        Tuple should contain field name and error message.
+        :rtype: list[tuple(str,str)]
+        """
+        pass
```

Comment:
Still think that this should be like the one I wrote in this comment https://github.com/shoopio/shoop/pull/457#discussion_r62484593 i.e. it doesn't return anything but rather is given the form and the hook can then add errors or do whatever it wants with the form.